source/mainnet/smart-contracts/best-practices/development.rst

```diff
@@ -0,0 +1,295 @@
+.. _sc-development-best-practices:
+
+==========================
+Development best practices
+==========================
+
+This document provides guidelines for developing smart contracts.
+It starts with some general thoughts about smart contract development and then gives more details about writing smart contracts in Rust for Concordium.
+
+Mindset
+========
+
+Smart contract development involves many risks that do not show up in, for example, web programming:
+
+- the cost of mistakes is very high;
+- possibilities for fixing bugs are limited;
+- the area is evolving constantly, with new vulnerabilities being discovered regularly;
+- malicious parties deliberately try to break your contract, for example, to steal the funds from the contract account.
+
+Therefore, it is not sufficient to defend your code against known vulnerabilities.
+You can think about smart contracts as mission-critical software, or software for embedded devices rather than a web application.
+
+To minimize the exposure of your smart contract to possible attacks consider the following.
+
+- Determine the “bare minimum” that should be on-chain: use smart contracts only for the part that requires decentralization.
+- Keep your contracts simple and minimalistic: stick to strictly necessary functionality only.
+  The more complexity you have in your contract, the larger the attack surface.
+- Be ready if things go wrong after deployment:
+
+  - provide “pause” functionality;
```

Comment:
We mention pausing several times without really describing it. Could you add a note about it or link to some example?

Reply:
I believe an example of pausing also addresses another issue in the documentation backlog: https://github.com/Concordium/concordium.github.io/issues/667

Reply:
No, that is a different thing enterily. The `interrupt`/`resume` events show up when a contract is invoking certain host functions. For example when it invokes another contract. When that happens, the caller is interrupted, then the callee is invoked (which may also produce events), and finally the caller is `resumed`.

Reply:
I've added a code snippet and pointed to the wCCD example